packages/vector_graphics_compiler/lib/src/paint.dart

```diff
@@ -140,6 +140,18 @@ class LinearGradient extends Shader {
 
   @override
   LinearGradient applyBounds(Rect bounds, AffineMatrix transform) {
+    if (unitMode == GradientUnitMode.userSpaceOnUse) {
```

Comment:
I'm fine with doing this in a separate patch, but I'd like to convert this to a bool, since that's really how it functions. It's really weird to have an enum compared by equality instead of swtich, and the enum only has two values, and really we just care about whether or not we need to transform the matrix or not.

Also, this code is missing applying `this.transform` if it's available. I'm not sure if I have an example handy where that would make a difference but I'm pretty sure it does.